src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/slick/WorkspaceComponent.scala

```diff
@@ -202,29 +206,69 @@ trait WorkspaceComponent {
     }
 
     def getAuthorizedRealms(workspaceIds: Seq[String], user: RawlsUserRef): ReadAction[Seq[Option[RawlsGroupRef]]] = {
-      DBIO.sequence(workspaceIds.map { id =>
-        findById(id) flatMap {
-          case None => DBIO.successful(None)
-          case Some(workspace) => DBIO.successful(workspace.realm)
-        }
-      }) flatMap { allRealms =>
+      (for {
+        workspace <- workspaceQuery if workspace.id.inSetBind(workspaceIds.map(UUID.fromString))
+      } yield workspace.realmGroupName).result flatMap { allRealms =>
         val flatRealms = allRealms.flatten.toSet
         DBIO.sequence(flatRealms.toSeq.map { realm =>
-          rawlsGroupQuery.loadGroupIfMember(realm, user) flatMap {
+          val groupRef = RawlsGroupRef(RawlsGroupName(realm))
+          rawlsGroupQuery.loadGroupIfMember(groupRef, user) flatMap {
             case None => DBIO.successful(None)
-            case Some(_) => DBIO.successful(Some(realm))
+            case Some(_) => DBIO.successful(Some(groupRef))
           }
         })
       }
     }
 
-    private def workspaceAttributes(workspaceId: UUID) = for {
-      workspaceAttrRec <- workspaceAttributeQuery if workspaceAttrRec.workspaceId === workspaceId
+    def listAccessGroupMemberEmails(workspaceIds: Seq[UUID], accessLevel: WorkspaceAccessLevel): ReadAction[Map[UUID, Seq[String]]] = {
```

Comment:
listAccessGroupMemberEmails and listSubmissionSummaryStats are new queries that load date for all workspaceIds at once (instead of per workspace as before). They also don't load data we don't actually need.